### PR TITLE
User API init

### PR DESCRIPTION
**Pull request by rathorevaibhav**
### Summary

1. Controller, schema (model), and migrations for User
2. User right now have only sample columns like first_name, last_name, email and phone. More columns will come in later
3. Boilerplate code for user test cases, fixtures, fallback controller etc that got added when ran the following command:
    ```sh
    mix phx.gen.json Users User user first_name:string last_name:string email:string phone:string
    ```
4. User REST API endpoint has been added to Postman

### Notes
1. Test cases will be worked on later after a basic workflow is in place

### CODE REVIEW

File: lib/portal/users.ex [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      iex> update_user(user, %{field: new_value})
+      {:ok, %User{}}
+
+      iex> update_user(user, %{field: bad_value})
+      {:error, %Ecto.Changeset{}}
+
+  """
+  def update_user(%User{} = user, attrs) do
+    user
+    |> User.changeset(attrs)
```
Comment (pritamps) [inline lib/portal/users.ex:72]: Do we need some checknig on the `attrs` here?

**Review by pritamps — CHANGES_REQUESTED** [review]
Hey @rathorevaibhav , considering the scope of this db, can we rename the repo to `af-db-backend`? Remove references to `portal` and repalce with `af-db` 

This looks cool! You will have to give a walkthrough of the code :)

File: lib/portal/users.ex [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  Returns the list of user.
+
+  ## Examples
+
+      iex> list_user()
+      [%User{}, ...]
+
+  """
+  def list_user do
+    Repo.all(User)
```
[the remark this answers is not in the document]
Comment (sudheshna-donthineni) [reply lib/portal/users.ex:21]: For analytics, we might need. Is there a downside to having it?

[the remark this answers is not in the document]
Comment (pritamps) [reply lib/portal/users.ex:21]: It doens't make sense to have a call that will return 10k-20k-1M users, no? WIll be very painful for the database. For analytics, we'll use BigQuery which will do this properly.

[the remark this answers is not in the document]
Comment (sudheshna-donthineni) [reply lib/portal/users.ex:21]: Makes sense. I keep forgetting about transactional vs analytics.

**Pull request closed without merging**